shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Validation.hs

```diff
@@ -40,13 +51,269 @@ import qualified Shelley.Spec.Ledger.STS.Bbody as STS
 import qualified Shelley.Spec.Ledger.STS.Chain as STS
 import Shelley.Spec.Ledger.STS.EraMapping ()
 import Shelley.Spec.Ledger.Slot (SlotNo)
+import Data.Proxy
+import Numeric.Natural
+import Data.Set (Set)
+import Data.Bifunctor
+import GHC.Exts as GHC
+import Shelley.Spec.Ledger.Genesis
+import qualified Data.Map as Map
+import Cardano.Ledger.Coin
 
 {-------------------------------------------------------------------------------
   Block validation API
 -------------------------------------------------------------------------------}
 
+
+newtype ModelTxId = ModelTxId Integer
```

Comment:
This is all in completely the wrong place... this is a testing concern

Reply:
New testing stuff should live in the `cardano-ledger-test` package

Reply:
These types are moved now.